src/software/SfM/main_ChangeLocalOrigin.cpp

```diff
@@ -0,0 +1,189 @@
+      if (argc == 1) throw std::string("Invalid command line parameter.");
+      cmd.process(argc, argv);
+  } catch(const std::string& s) {
+      std::cerr << "Usage: " << argv[0] << '\n'
+        << "[-i|--input_file] path to the input SfM_Data scene\n"
+        << "[-o|--output_dir] path to the output SfM_Data scene (in local frame)\n"
+        << "[-l|--local_frame_origin] \"x;y;z\" of local frame origin\n"
+        << "[-f|--first_frame_origin] use position of first frame as origin\n"
+        << std::endl;
+
+      std::cerr << s << std::endl;
+      return EXIT_FAILURE;
+  }
+
+  if (sOutDir.empty())
+  {
+    std::cerr << std::endl
+      << "No output SfM_Data filename specified." << std::endl;
+    return EXIT_FAILURE;
+  }
+  
+
+  // Load input SfM_Data scene
+  SfM_Data sfm_data;
+  if (!Load(sfm_data, sSfM_Data_Filename_In, ESfM_Data(ALL)))
+  {
+    std::cerr << std::endl
+      << "The input SfM_Data file \"" << sSfM_Data_Filename_In << "\" cannot be read." << std::endl;
+    return EXIT_FAILURE;
+  }
+  
+  // Local origin
+  b_first_frame_origin = cmd.used('f');
+  
+  if (sLocalFrameOrigin.empty() && !b_first_frame_origin)
+  {
+    std::cerr << std::endl
+      << "No local frame origin specified." << std::endl;
+    return EXIT_FAILURE;
+  }
+  else
+  {
+    if (b_first_frame_origin)
+    {
+      local_Frame_Origin =  (sfm_data.poses.begin()->second).center();
+      std::cout<<"Using frame origin: "<<local_Frame_Origin<<"\n";
+    }
+    else
+    {
+      std::vector<std::string> vec_str;
+      stl::split(sLocalFrameOrigin, ';', vec_str);
+      if (vec_str.size() != 3)
+      {
+        std::cerr << "\n Missing ';' character in local frame origin" << std::endl;
+        return false;
+      }
+      // Check that all local frame origin values are valid numbers
+      for (size_t i = 0; i < vec_str.size(); ++i)
+      {
+        double readvalue = 0.0;
+        std::stringstream ss;
+        ss.str(vec_str[i]);
+        if (! (ss >> readvalue) )  {
+          std::cerr << "\n Used an invalid not a number character in local frame origin" << std::endl;
+          return false;
+        }
+        local_Frame_Origin[i] = readvalue;
+      }
+    }
+  }
+  
+  // Move poses
+  {
+    C_Progress_display my_progress_bar(sfm_data.poses.size(),
+                                       std::cout,
+                                       "\nRecalculating poses:\n");
+    // Loop through views and structure and correct the position according to local frame
+    for (Poses::iterator itPose = sfm_data.poses.begin(); itPose != sfm_data.poses.end(); ++itPose)
+    {
+      Pose3 & pose = itPose->second;   
+      pose = Pose3(pose.rotation(), pose.center()-local_Frame_Origin);
+      ++my_progress_bar;
+    }
+  }
+    
+  // Move view priors
+  {
+    for (auto & view : sfm_data.GetViews())
+    {
+      if (sfm::ViewPriors *prior = dynamic_cast<sfm::ViewPriors*>(view.second.get()))
+      {
+	    if (prior->b_use_pose_center_)
```

Comment:
tab inconsistency. But as I said in the general comments, I think it can be better to use the `ApplySimiliarity` function ;-) to apply a given transform (here a translation) to the whole sfm_data object.